KAFKA-3762 Log.loadSegments() should log the message in exception

imandhan opened the pull request:
Adding an error logging message in Log.loadSegments() in the case when an index file corresponding to a log file exists but an exception is thrown.

Signed-off-by: Ishita Mandhan imandha@us.ibm.com

imandhan committed:
KAFKA-3762 Log.loadSegments() should log the message in exception

Adding an error logging message in Log.loadSegments() in the case when an index file corresponding to a log file exists but an exception is thrown.

Signed-off-by: Ishita Mandhan <imandha@us.ibm.com>

ewencp commented on core/src/main/scala/kafka/log/Log.scala:
```diff
@@ -197,6 +197,7 @@ class Log(val dir: File,
               warn("Found a corrupted index file, %s, deleting and rebuilding index...".format(indexFile.getAbsolutePath))
               indexFile.delete()
               segment.recover(config.maxMessageSize)
+              error("Corrupted Index File", e)
```
If we already have a `warn` above, why do we need a separate `error`? If the exception and stack trace need to be logged, can we just incorporate it into the previous warning (and change to error if we think it needs to be at that log level)?

imandhan replied:
makes sense...will update the above warning to an error

ijuma replied:
The idea was to add the message from the exception to the `warn` statement in line 197.

imandhan committed:
no message

Signed-off-by: Ishita Mandhan <imandha@us.ibm.com>

imandhan committed:
KAFKA-3762 Log.loadSegments() should log the message in exception

Adding an error logging message in Log.loadSegments() in the case when an index file corresponding to a log file exists but an exception is thrown.

Signed-off-by: Ishita Mandhan <imandha@us.ibm.com>

ijuma commented on core/src/main/scala/kafka/log/Log.scala:
```diff
@@ -194,7 +194,7 @@ class Log(val dir: File,
           } catch {
             case e: java.lang.IllegalArgumentException =>
-              warn("Found a corrupted index file, %s, deleting and rebuilding index...".format(indexFile.getAbsolutePath))
+              warn("Found a corrupted index file, %s, deleting and rebuilding index.".format(indexFile.getAbsolutePath), e)
```
Does the stack trace add any value here? I thought that Jun was only asking for the exception message to be included. cc @junrao

junrao replied:
Yes, I think including the message, but not the stacktrace in the exception is enough.

imandhan replied:
Okay, I'll make the change. Thanks for bearing with me!

imandhan committed:
KAFKA-3762 Log.loadSegments() should log the message in exception

Adding an error logging message in Log.loadSegments() in the case when an index file corresponding to a log file exists but an exception is thrown.

ewencp commented:
LGTM
